### PR TITLE
isFeatureEnabled now returns false if disabled

### PR DESCRIPTION
**Pull request by marandaneto**
**What does this PR do?**

`isFeatureEnabled` should return `false` if feature flag is disabled

**Where should the reviewer start?**
`PHGPostHog.m`

**How should this be manually tested?**
Demo project has one flag for each case, you can just call `isFeatureEnabled` with them.

**Any background context you want to provide?**

Relates to https://github.com/PostHog/posthog-android/pull/42

**What are the relevant tickets?**

**Screenshots or screencasts (if UI/UX change)**

**Questions:**
- Does the docs need an update?
- Are there any security concerns?
- Do we need to update engineering / success?

### CODE REVIEW

File: Examples/CarthageExample/CarthageExample/AppDelegate.swift [hunk 1]
```diff
@@ -2,7 +2,7 @@ import UIKit
 import PostHog
 
 // Use your own apiKey!
-let PostHog = PHGPostHog(configuration: PHGPostHogConfiguration(apiKey: "b2XoQDHA5YhWQAjr2FZ4qIckgyEqXyY8"))
+let PostHog = PHGPostHog(configuration: PHGPostHogConfiguration(apiKey: "_6SG-F7I1vCuZ-HdJL3VZQqjBlaSb1_20hDPwqMNnGI"))
```
Comment (marandaneto) [inline Examples/CarthageExample/CarthageExample/AppDelegate.swift:5]: Defaults to "Demo".

File: PostHogTests/FeatureFlagTests.swift [hunk 2]
```diff
@@ -27,13 +27,24 @@ class FeatureFlagTests: QuickSpec {
     it("checks flag is enabled") {
       _ = stubRequest("POST", "https://app.posthog.test/decide/?v=3" as LSMatcheable)
         .andReturn(200)?
-        .withBody("{\"featureFlags\":{\"some-flag\":\"true\"}}" as LSHTTPBody);
+        .withBody("{\"featureFlags\":{\"some-flag\": true}}" as LSHTTPBody);
```
Comment (marandaneto) [inline PostHogTests/FeatureFlagTests.swift:30]: String escaping was making the test wrong, fixed all of them.

File: PostHogTests/PostHogTests.swift [hunk 3]
```diff
@@ -97,22 +97,22 @@ class PostHogTests: QuickSpec {
 
     it("fires Application Opened during UIApplicationWillEnterForeground") {
       testMiddleware.swallowEvent = true
-      NotificationCenter.default.post(name: UIApplication.willEnterForegroundNotification, object: testApplication)
+      NotificationCenter.default.post(name: NSNotification.Name.UIApplicationWillEnterForeground, object: testApplication)
```
Comment (marandaneto) [inline PostHogTests/PostHogTests.swift:100]: `UIApplication.x` is deprecated in favor of `NSNotification.Name.x`

File: PostHog/Classes/PHGPostHog.m [hunk 4]
```diff
@@ -313,8 +313,18 @@ - (bool)isFeatureEnabled:(NSString *)flagKey
...
+    bool isFlagEnabled = true;
+    id value = [flags valueForKey:flagKey];
+    
+    if (value != nil) {
+        if ([value isKindOfClass:[NSNumber class]]) {
+            isFlagEnabled = [value boolValue];
+        }
+    } else {
+        isFlagEnabled = false;
+    }
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply PostHog/Classes/PHGPostHog.m:327]: @benjackwhite context is her https://github.com/PostHog/posthog-android/pull/42#issuecomment-1751966655
If a flag is `true` or any string, it returns `true`.
if a flag is `false`, it should return `false`, which is not the case before this change.

[the remark this answers is not in the document]
Comment (benjackwhite) [reply PostHog/Classes/PHGPostHog.m:327]: Ah fair enough with that context. It was more to say "currently the API never returns `false` - it just isn't defined"

**Review by benjackwhite — APPROVED** [review]

**Pull request merged**